src/mscorlib/shared/System/SpanHelpers.Char.cs

```diff
@@ -167,6 +168,90 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
             }
         }
 
+        public static unsafe int LastIndexOf(ref char searchSpace, char value, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            fixed (char* pChars = &searchSpace)
+            {
+                char* pCh = pChars + length;
+                char* pEndCh = pChars;
+
+#if !netstandard11
+                if (Vector.IsHardwareAccelerated && length >= Vector<ushort>.Count * 2)
+                {
+                    const int elementsPerByte = sizeof(ushort) / sizeof(byte);
+                    length = (((int)pCh & (Unsafe.SizeOf<Vector<ushort>>() - 1)) / elementsPerByte) + 1;
```

Comment:
A case I think this is missing is when `pCh` is already at the correct position for alignment.  With the current code, we will do a whole SequentialScan on the first vector's length of chars before we start the vectorized section.  We can skip that SequentialScan when `pCh` is already at the correct position for alignment.

Using my back of the napkin pseudo code, I think you can accomplish that with adding the line:

`length = length & (Vector<ushort>.Count - 1);`

This is also what the SpanHelpers.Byte.LastIndexOf version is also doing - [https://github.com/dotnet/coreclr/blob/d1f49cc4b26d28fdb1c4f3bf147668e13b8995eb/src/mscorlib/shared/System/SpanHelpers.Byte.cs#L261](https://github.com/dotnet/coreclr/blob/d1f49cc4b26d28fdb1c4f3bf147668e13b8995eb/src/mscorlib/shared/System/SpanHelpers.Byte.cs#L261)

Reply:
If `pCh` is already aligned, adding that line won't skip the SequentialScan (due to the + 1):
```C#
// ((int)pCh & (Unsafe.SizeOf<Vector<ushort>>() - 1)) = 0
// 0 / elementsPerByte = 0
// length =  0 + 1 = 1
length = (((int)pCh & (Unsafe.SizeOf<Vector<ushort>>() - 1)) / elementsPerByte) + 1; 

length = length & (Vector<ushort>.Count - 1); // length = 1

```

Reply:
It's not `pCh` that needs to be aligned.  It's `pStart` below. Like I said above, the case where `pCh` is at `the correct position for alignment`.


```
A B C D | E F G H | I J K
          ^     ^
          |     pCh
          pStart
```

It's pStart that needs to be aligned.

So you are checking for when pCh is at the end of the alignment.  i.e. `length = Vector<ushort>.Count`.